### PR TITLE
fix(toolbar): fix allowed origins form saving

### LINKED ISSUE #78689: [DevToolbar] 'Allowed-domains' project settings does not persist
Repro Steps:
1. Visit the Dev Toolbar settings page for a project where you have permissions(such as https://sentry.sentry.io/settings/projects/javascript/toolbar/)
2. Enter a value in the "Allowed Origins" textbox. Anything, for example `localhost`
3. Click Save
4. Notice that the 'save' button is disabled now. In the browser network inspector a 200 request was completed.
5. Reload the page
6. Notice that the value you entered is gone!

### PR DESCRIPTION
**Pull request by aliu39**
Closes https://github.com/getsentry/sentry/issues/78689

Also adds a jest test and renames file/class to match other project settings (`project` prefix)

### CODE REVIEW

File: static/app/views/settings/project/projectToolbar.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  const {routerProps, organization, project, router} = initializeOrg();
+  const url = `/projects/${organization.slug}/${project.slug}/`;
+
+  beforeEach(function () {
+    MockApiClient.clearMockResponses();
+    MockApiClient.addMockResponse({
+      url,
+      method: 'GET',
+      body: ProjectFixture(),
+    });
```
Comment (billyvg) [inline static/app/views/settings/project/projectToolbar.spec.tsx:18]: When/where does this get called? should we add an expectation for it?

File: static/app/views/settings/project/projectReplays.tsx [hunk 2]
```diff
@@ -80,6 +81,9 @@ function ProjectReplaySettings({organization, project, params: {projectId}}: Pro
         apiMethod="PUT"
         apiEndpoint={`/projects/${organization.slug}/${projectId}/`}
         initialData={project.options}
+        onSubmitSuccess={(
+          response // This will update our project context
+        ) => ProjectsStore.onUpdateSuccess(response)}
```
Comment (billyvg) [inline static/app/views/settings/project/projectReplays.tsx:86]: Does this work?

```suggestion
        onSubmitSuccess={ProjectsStore.onUpdateSuccess}
```

**Review by billyvg — COMMENTED** [review]
How do we test that the project store gets updated properly since that seems to be the source of the bug?

File: static/app/views/settings/project/projectToolbar.spec.tsx (hunk 1 above)
Comment (aliu39) [reply static/app/views/settings/project/projectToolbar.spec.tsx:18]: Adding a test for this!

File: static/app/views/settings/project/projectReplays.tsx (hunk 2 above)
Comment (ryan953) [reply static/app/views/settings/project/projectReplays.tsx:86]: won't :(

https://github.com/getsentry/sentry/blob/d81a027147e7f97891f4e9f921d766b9912aded1/static/app/stores/projectsStore.tsx#L101-L103

**Review by ryan953 — APPROVED** [review]

File: static/app/views/settings/project/projectReplays.tsx (hunk 2 above)
Comment (aliu39) [reply static/app/views/settings/project/projectReplays.tsx:86]: why not? @ryan953

File: static/app/views/settings/project/projectToolbar.spec.tsx (hunk 1 above)
Comment (aliu39) [reply static/app/views/settings/project/projectToolbar.spec.tsx]: updated the tests. We don't need to mock this method for this component

**Comment by aliu39** [comment]
The source of the bug was using the "Save Changes" button for the parent `<Form>` component, rather than `saveOnBlur` for each individual field in the form group. Seems like the button doesn't have access to the fields' `getData` prop which we use the wrap the data with `options: {}` before making the api request. 

Tldr; we were sending the wrong data with `{sentry:toolbar_allowed_origins: ...}` instead of `{options: {sentry:toolbar_allowed_origins: ...}}`

File: static/app/views/settings/project/projectReplays.tsx (hunk 2 above)
Comment (ryan953) [reply static/app/views/settings/project/projectReplays.tsx:86]: it's not using an arrow function expression to bind the `this` keyword, but it uses `this` inside the function body. Therefore we must use dot-syntax to bind the value of `this`.
